File: Javascript Implementation/toxic.js
const Discord = require('discord.js');
const { LanguageServiceClient } = require('@google-cloud/language'); //Google's Perspective AI

const client = new Discord.Client();
const languageClient = new LanguageServiceClient();
const prefix = '!toxic_bot';

client.on('ready', () => {
  console.log(`Logged in as ${client.user.tag}!`);
});

client.on('message', async (message) => {
  if (message.content.startsWith(prefix)) {
    const userMessage = message.content.slice(prefix.length).trim();

    const document = {
      content: userMessage,
      type: 'PLAIN_TEXT',
    };
    const [result] = await languageClient.analyzeSentiment({
      document: document,
      encodingType: 'UTF8',
    });

    // toxic %
    const toxicityPercentage = (result.documentSentiment.score * 100).toFixed(2);

    message.channel.send(`The toxicity percentage of your message is ${toxicityPercentage}%.`);
  }
});

client.login('DISCORD_BOT_TOKEN');

File: C# Implementation/toxic.cs
using Discord;
using Discord.WebSocket;
using Google.Cloud.Language.V1;
using System;
using System.Threading.Tasks;

namespace ToxicBot
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var discordClient = new DiscordSocketClient();
            var languageClient = LanguageServiceClient.Create();
            var prefix = "!toxic_bot";

            discordClient.Log += Log;

            discordClient.MessageReceived += async (message) =>
            {
                if (!message.Author.IsBot && message.Content.StartsWith(prefix))
                {
                    var userMessage = message.Content.Substring(prefix.Length).Trim();

                    // perspective ai
                    var document = new Document
                    {
                        Content = userMessage,
                        Type = Document.Types.Type.PlainText
                    };
                    var response = await languageClient.AnalyzeSentimentAsync(new AnalyzeSentimentRequest
                    {
                        Document = document,
                        EncodingType = EncodingType.Utf8
                    });

                    // toxic %
                    var toxicityPercentage = Math.Round(response.DocumentSentiment.Score * 100, 2);

                    await message.Channel.SendMessageAsync($"The toxicity percentage of your message is {toxicityPercentage}%.");
                }
            };

            await discordClient.LoginAsync(TokenType.Bot, "DISCORD_BOT_TOKEN");
            await discordClient.StartAsync();

            await Task.Delay(-1);
        }

        private static Task Log(LogMessage message)
        {
            Console.WriteLine(message.ToString());
            return Task.CompletedTask;
        }
    }
}
